server/test/controllers/applicant/ApplicantProgramsControllerTest.java

```diff
@@ -433,6 +433,14 @@ public void edit_redirectsToFirstIncompleteBlock() {
                 .url());
   }
 
+  @Test
+  public void hxFilter_isOk() {
+    Result result =
+        controller.hxFilter(fakeRequest(), ImmutableList.of()).toCompletableFuture().join();
+
+    assertThat(result.status()).isEqualTo(OK);
```

Comment:
(blocking) After adding the exceptionally in the controller please try to test the failure case.

Reply:
As we discussed, I wasn't able to cause an exception in the unit test.  I did test manually, however, and verified that we now log an error message server-side and, client-side, HTMX catches the error, triggers its own error event and doesn't swap the partial.